cmd/object-handlers.go

```diff
@@ -88,6 +89,41 @@ func (f funcToWriter) Write(p []byte) (int, error) {
 	return f(p)
 }
 
+// HTTPWriteCloser writes to the underlying io.Writer
+// and also writes nil to the underlying io.Writer when
+// Close is called and no data was written.
+//
+// In case of an http.ResponseWriter the Close will set the
+// HTTP headers depending upon whether no data was written to the client.
+type HTTPWriteCloser struct {
```

Comment:
We should move this into a package of its own @aead and make it generic of sort.  Not just `HTTPWriteCloser` ?

Reply:
We can - `pkg/ioutil`?! I can also take a look whether the functionality is already implemented somewhere but maybe a custom `ioutil` package is not a bad idea - there's probably more io related stuff we can extract an reuse...